cmake/python_functions.cmake

```diff
@@ -0,0 +1,30 @@
+#############################################################################
+# Copyright (c) 2018 Balabit
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+find_program(PEP8_EXECUTABLE NAMES pep8 pycodestyle HINTS ${CMAKE_BINARY_DIR}/.venv/bin)
+find_program(NOSETESTS_EXECUTABLE nosetests HINTS ${CMAKE_BINARY_DIR}/.venv/bin)
+find_program(PYLINT_EXECUTABLE pylint HINTS ${CMAKE_BINARY_DIR}/.venv/bin)
+
+
+add_custom_target(python-venv
+	COMMAND rm -rf .venv && virtualenv -p ${PYTHON_EXECUTABLE} .venv && .venv/bin/pip install -r ${CMAKE_HOME_DIRECTORY}/requirements.txt)
```

Comment:
I just wanted to try this out from a clean environment. In my case there is no .venv directory, so `make python-venv` failed.

Does `virtualenv` work with python3? As far as I know, for python3, we should use `pyvenv`. This should be a problem in the autotools version as well.

Reply:
rm -rf should remove .venv without an error if it doesn't exist. virtualenv does work for python3 as well, yes.

Reply:
You are right! The problem was not with rm -rf, but with `${PYTHON_EXECUTABLE}`:
```
The path .venv (from --python=.venv) is not an executable file
```
Probably the executable detection happens in a later phase, so `${PYTHON_EXECUTABLE}` is an empty string, then virtualenv fails because she tries to execute `virtualenv -p .venv`.

Right, thanks for explaining virtualenv and python3. I was confused because python3-virtualenv on ubuntu does not ship the virtualenv executable, that's why I was thinking they are incompatible. If I pass python3 binary to -p, it works indeed.